### PR TITLE
Adding GlobalAveragePooling 3D

### PR DESCRIPTION
**Pull request by Shashi456**

### CODE REVIEW

File: Sources/DeepLearning/Layer.swift [hunk 1]
```diff
@@ -1022,7 +1022,7 @@ public struct GlobalAveragePooling1D<Scalar: TensorFlowFloatingPoint>: Layer {
     ///   - input: The input to the layer.
     ///   - context: The contextual information for the layer application, e.g. the current learning
     ///     phase.
-    /// - Returns: The output 
+    /// - Returns: The output
```
Comment (rxwei) [inline Sources/DeepLearning/Layer.swift]: Add period.
```suggestion
    /// - Returns: The output.
```

File: Sources/DeepLearning/Layer.swift [hunk 2]
```diff
@@ -1048,6 +1048,25 @@ public struct GlobalAveragePooling2D<Scalar: TensorFlowFloatingPoint>: Layer {
...
+    /// Creates a global average pooling layer.
+    public init() {}
+
+    /// Returns the output obtained from applying the layer to the given input.
+    ///
+    /// - Parameters:
+    ///   - input: The input to the layer.
+    ///   - context: The contextual information for the layer application, e.g. the current learning
+    ///     phase.
+    /// - Returns: The output
```
Comment (rxwei) [inline Sources/DeepLearning/Layer.swift]: Add period.

File: Sources/DeepLearning/Layer.swift [hunk 3]
```diff
@@ -1048,6 +1048,25 @@ public struct GlobalAveragePooling2D<Scalar: TensorFlowFloatingPoint>: Layer {
...
+    @differentiable
+    public func applied(to input: Tensor<Scalar>, in _: Context) -> Tensor<Scalar> {
+        return input.mean(alongAxes: [1,2,3]).reshaped(to: [input.shape[0], input.shape[4]])
```
Comment (rxwei) [inline Sources/DeepLearning/Layer.swift]: Add a space after each comma.

```suggestion
        return input.mean(alongAxes: [1, 2, 3]).reshaped(to: [input.shape[0], input.shape[4]])
```

**Comment by Shashi456** [comment]
@rxwei is it okay if i add the tests in another PR? I think most of the conv layers lack a test, So i'll go through them one by one?

**Comment by rxwei** [comment]
Sure. Sounds good.

**Pull request merged**